### PR TITLE
#235 Added pestertests and input validation for publish-bicep

### PR DESCRIPTION
**Pull request by bjompen**

### CODE REVIEW

**Comment by bjompen** [comment]
@StefanIvemo Can you please verify the testcases that should work, and that we are covering all different approved inputs?
The verification will check:
String starts with "br followed by : or /" 
String ends with ": followed by at least one alphanumeric char"

**Comment by StefanIvemo** [comment]
It looks good and works as expected. While testing I found a pattern that passed validation but failed on `Publish-BicepNetFile`.

```powershell
Publish-Bicep -Path C:\temp\bicep\storage.bicep -Target br/myModules/steffes/storrage:v33
Publish-BicepNetFile: C:\git\stefanivemo\PSBicep\Source\Public\Publish-Bicep.ps1:39:13
Line |
  39 |              Publish-BicepNetFile -Path $BicepFile.FullName -Target $T …
     |              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | The module alias name "myModules/steffes/storrage" is invalid. Valid characters are
     | alphanumeric, "_", or "-".
```

Can we verify if `-Target` starts with `br/` (user is refering to an alias), that `.azurecr.io` is not part of the alias name?

**Comment by StefanIvemo** [comment]
I guess the tests looks ok. In my world all tests are good if they pass 😊. @JohnRoos can you review the tests?

**Comment by bjompen** [comment]
improved the regex so it verifies and checks for FQDN (azurecr.io) or modulealias.

File: Source/Public/Publish-Bicep.ps1 [hunk 1]
```diff
@@ -8,6 +8,7 @@ function Publish-Bicep {
 
         [Parameter(Mandatory, Position = 2)]
         [ValidateNotNullOrEmpty()]
+        [ValidatePattern('^(?<Prefix>[bBrR]{2})(?<ACROrAlias>(:[\w\-_]+\.azurecr.io|\/[\w\-\._]+:))(?<path>[\w\/\-\._]+)(?<tag>:[\w\/\-\._]+)$', ErrorMessage = 'Target does not match pattern for registry.')]
```
Comment (StefanIvemo) [inline Source/Public/Publish-Bicep.ps1:11]: Can we change the errorMessage to give the user a better understanding on why validation is failing?

`ErrorMessage = 'Target does not match pattern for registry. Specify a path to a registry using "br:", or "br/" if using an alias.'`

Comment (bjompen) [reply Source/Public/Publish-Bicep.ps1]: done!

**Review by StefanIvemo — APPROVED** [review]
LGTM! 🚀

**Pull request merged**